crates/holochain_conductor_api/src/app_interface.rs

```diff
@@ -254,89 +271,36 @@ pub enum AppResponse {
     Ok,
 }
 
-/// The data provided over an app interface in order to make a zome call
+/// The data provided over an app interface in order to make a zome call.
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct ZomeCall {
-    /// The ID of the cell containing the zome to be called
-    pub cell_id: CellId,
-    /// The zome containing the function to be called
-    pub zome_name: ZomeName,
-    /// The name of the zome function to call
-    pub fn_name: FunctionName,
-    /// The serialized data to pass as an argument to the zome function call
-    pub payload: ExternIO,
-    /// The capability request authorization
-    ///
-    /// This can be `None` and still succeed in the case where the function
-    /// in the zome being called has been given an `Unrestricted` status
-    /// via a `CapGrant`. Otherwise it will be necessary to provide a `CapSecret` for every call.
-    pub cap_secret: Option<CapSecret>,
-    /// The provenance (source) of the call
-    /// MUST match the signature.
-    pub provenance: AgentPubKey,
+pub struct ZomeCallParamsSigned {
+    /// Bytes of the serialized zome call payload that consists of all fields of the
+    /// [`ZomeCallParams`].
+    pub bytes: ExternIO,
+    /// Signature by the provenance of the call, signing the bytes of the zome call payload.
     pub signature: Signature,
-    pub nonce: Nonce256Bits,
-    pub expires_at: Timestamp,
 }
 
-impl From<ZomeCall> for ZomeCallUnsigned {
-    fn from(zome_call: ZomeCall) -> Self {
+impl ZomeCallParamsSigned {
+    pub fn new(bytes: Arc<[u8]>, signature: Signature) -> Self {
         Self {
-            cell_id: zome_call.cell_id,
-            zome_name: zome_call.zome_name,
-            fn_name: zome_call.fn_name,
-            payload: zome_call.payload,
-            cap_secret: zome_call.cap_secret,
-            provenance: zome_call.provenance,
-            nonce: zome_call.nonce,
-            expires_at: zome_call.expires_at,
+            bytes: ExternIO::from(bytes.to_vec()),
+            signature,
         }
     }
-}
 
-impl ZomeCall {
-    pub async fn try_from_unsigned_zome_call(
+    pub async fn try_from_params(
         keystore: &MetaLairClient,
-        unsigned_zome_call: ZomeCallUnsigned,
+        params: ZomeCallParams,
     ) -> LairResult<Self> {
-        let signature = unsigned_zome_call
+        let bytes = params.serialize().map_err(|e| e.to_string())?;
+        // Signature is generated for the hash of the serialized zome call params.
+        let bytes_hash = blake2b_256(&bytes);
```

Comment:
I'd still like to recommend sha2_512--since we're breaking the api anyways, this would give us some additional bytes to sign over for more security, and sha2 is a library people are more used to having to import, so it makes the clients more obvious.

If we're against making this change for some reason, I'm also okay as-is, just a suggestion.

Reply:
I'm fine changing to sha2-512.

Reply:
@ThetaSinner Does that mean you're against changing to sha2-512?

Reply:
@neonphog Switched to SHA2 512-bit.